src/SqlPersistence/Outbox/SqlDialect_MsSqlServer.cs

```diff
@@ -80,6 +80,24 @@ internal override string GetOutboxCleanupCommand(string tableName)
 where Dispatched = 'true' and
       DispatchedAt < @DispatchedBefore";
             }
+
+            internal override string AddOutboxPadding(string json)
+            {
+                //We need to ensure the outbox content is at lest 8000 bytes long because otherwise SQL Server will attempt to
+                //store is inside the data page which will result in low space utilization.
+
+                //We use 4000 instead of 8000 in the condition because the SQL Persistence uses nvarchar data type which encodes
+                //strings at UTF-16 (2 bytes per character)
+
+                //We allow content smaller than 1800 characters (3600 bytes) to not be padded because such content does not block
```

Comment:
We figure them out via experiments. Well, I believe the 4000 value is built-in to the SQL Server query optimization (half the page size). The 3600 comes from the fact that there is some (between 100 and 400) data in the outbox row _in addition_ to the messages values. So we wanted to be on the safe side and allocated as much as 400 bytes in case in future we add more constant-size columns to the table.

Reply:
Can we add some doco about this somwhere? Either in the Platform Repo or in a docs folder in this repo?

Reply:
Added here https://github.com/Particular/Platform/commit/e09cdf3b6e12aa7ecb7e82cda885ee77e323eaa9